resources/lib/kodiutils.py

```diff
@@ -385,9 +368,11 @@ def localize_from_data(name, data):
 
 def get_setting(key, default=None):
     ''' Get an add-on setting '''
-    from xbmcaddon import Addon
+    global ADDON  # pylint: disable=global-statement
+    from xbmcaddon import Addon # pylint: disable=redefined-outer-name,reimported
```

Comment:
Re-importing `Addon` is not necessary. Unless you have a good reason.

Reply:
That was a test, in my running code I don't do it anymore because it is painfully slow.

Reply:
We need to test if the settings now affect the menu's instantly, because that was the reason for doing this.

Reply:
I tested. Re-assigning the global variable is still needed, but you can remove `from xbmcaddon import Addon`

Reply:
Sure, there's no need to reimport it. But the reimport is not causing the performance regression. Re-instantiating Addon is causing the regression, so that is not something we want to do when doing a `get_setting` but we ought to do this on an `onSettingsChanged` event.

So that's the issue IMO. A solution may be to cache this in window properties, where `get_properties()` would read from the window property cache, and service.py would write to the window property cache when an `onSettingsChanged` event happens.

It means we have to iterate over the known settings in service.py.